pkgs/build-support/setup-hooks/auto-patchelf.py

```diff
@@ -0,0 +1,332 @@
+    for dep in file_dependencies:
+        if dep.is_absolute() and dep.is_file():
+            # This is an absolute path. If it exists, just use it.
+            # Otherwise, we probably want this to produce an error when
+            # checked (because just updating the rpath won't satisfy
+            # it).
+            continue
+        elif (libc_lib / dep).is_file():
+            # This library exists in libc, and will be correctly
+            # resolved by the linker.
+            continue
+
+        if foundDependency := findDependency(dep.name, file_arch, file_osabi):
+            rpath.append(foundDependency)
+            dependencies.append(Dependency(path, dep, True))
+            print(f"    {dep} -> found: {foundDependency}")
+        else:
+            dependencies.append(Dependency(path, dep, False))
+            print(f"    {dep} -> not found!")
+
+    # Dedup the rpath
+    rpath = ":".join(dict.fromkeys(map(Path.as_posix, rpath)))
+
+    if rpath:
+        print("setting RPATH to:", rpath)
+        subprocess.run(
+                ["patchelf", "--set-rpath", rpath, path.as_posix()],
+                check=True)
+
+    return dependencies
+
+
+def autoPatchelf(
+        pathsToPatch,
+        libDirs,
+        runtime_deps,
+        recursive=True,
+        ignoreMissing=False):
+
+    if not pathsToPatch:
+        sys.exit("No paths to patch, stopping.")
+
+    # Add all shared objects of the current output path to the cache,
+    # before libDirs, so that they are chosen first in findDependency.
+    populate_cache(pathsToPatch, recursive)
+    populate_cache(libDirs)
+
+    dependencies = []
+    for path in chain.from_iterable(glob(p, '*', recursive) for p in pathsToPatch):
+        if not path.is_symlink() and path.is_file():
+            dependencies += autoPatchelfFile(path, runtime_deps)
+
+    # Print a summary of the missing dependencies at the end
+    for dep in dependencies:
+        if not dep.found:
+            print(f"auto-patchelf could not satisfy dependency {dep.name} wanted by {dep.file}")
+
+    any_missing = any(not dep.found for dep in dependencies)
+    if any_missing and not ignoreMissing:
+        sys.exit('auto-patchelf failed to find all the required dependencies.\n'
+                 'Add the missing dependencies to --libs or use --ignore-missing.')
+
+    some_found = any(dep.found for dep in dependencies)
+    if not some_found:
+        print('auto-patchelf failed to find any file to patch.\n'
+              'It may have been misconfigured, or you may not need it.')
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog="auto-patchelf",
+        description='auto-patchelf tries as hard as possible to patch the'
+                    ' provided binary files by looking for compatible'
+                    'libraries in the provided paths.')
+    parser.add_argument(
+        "--ignore-missing",
+        dest="ignoreMissing",
```

Comment:
Btw. you don't need those, since `argparse` by default will generate PEP-8 compatible attributes. In this case you'll get `ignore_missing`.